tracer/src/Datadog.Trace/Ci/CIVisibility.cs

```diff
@@ -425,48 +427,84 @@ internal static IApiRequestFactory GetRequestFactory(ImmutableTracerSettings set
         internal static IApiRequestFactory GetRequestFactory(ImmutableTracerSettings tracerSettings, TimeSpan timeout)
         {
             IApiRequestFactory? factory = null;
-
+            var exporterSettings = tracerSettings.ExporterInternal;
+            var strategy = exporterSettings.TracesTransport;
+
+            switch (strategy)
+            {
+                case TracesTransportType.WindowsNamedPipe:
+                    Log.Information<string?, int>("Using " + nameof(NamedPipeClientStreamFactory) + " for CI Visibility transport, with pipe name {TracesPipeName} and timeout {TracesPipeTimeoutMs}ms.", exporterSettings.TracesPipeNameInternal, exporterSettings.TracesPipeTimeoutMsInternal);
```

Comment:
Is this identical to the default Agent strategy? Can't you use `AgentTransportStrategy` if so? What's different here? 🤔

Reply:
The AutomaticDescompression part and the custom timeout I think.

Reply:
Changed in https://github.com/DataDog/dd-trace-dotnet/pull/5634/commits/8a129d82b67a6b57b7e720c8c787405c8d8bd876 to reuse the strategy if is not tcp

Reply:
Could I suggest an alternative? Just add support for the automatic decompression and custom timeout to the default strategy?